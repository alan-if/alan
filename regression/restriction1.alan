Syntax x = x (y)
  Where y Isa object
    Else "Should be an object."

Add To Every actor
  Definite Article
  Indefinite Article
End Add.

Add To Every object
  Verb x
    Does
      "Was an object."
  End Verb x.
End Add.

The o Isa object At l.
End The o.

The a Isa actor At l.
End The a.

The l Isa location.
End The l.

Start At l.
