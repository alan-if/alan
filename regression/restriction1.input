x the o
x the a

