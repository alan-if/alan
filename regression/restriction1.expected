########## restriction1 ##########

        No warnings or errors detected.



L
There is a o here. There is a here.

> x the o
Was an object.

> x the a
Should be an object.

> 

> 

Do you want to UNDO, RESTART, RESTORE or QUIT ? 
